include/pybind11/detail/internals.h

```diff
@@ -11,6 +11,8 @@
 
 #include "../pytypes.h"
 
+#include <exception>
```

Comment:
I'm putting these in the "standard" location, at the top of the file they are used in, rather than bunched together and included via other files. Assuming we use include-what-you-use, the rest of the includes will show up here eventually.